android-lib/src/androidTest/java/com/kinvey/androidTest/store/DeltaCacheTest.java

```diff
@@ -0,0 +1,461 @@
+        assertEquals("DeltaCacheUserName", callback.getResult().getUsername());
+
+        CustomKinveySyncCallback<Person> syncCallback = testManager.sync(store, client.query());
+        assertNotNull(syncCallback);
+        assertNotNull(syncCallback.getResult());
+        assertNull(syncCallback.getError());
+        assertNotNull(syncCallback.getResult().getResult().get(0).getUsername());
+        assertEquals("DeltaCacheUserName", syncCallback.getResult().getResult().get(0).getUsername());
+    }
+
+    @Test
+    public void testCreateSync() throws InterruptedException, IOException {
+        client.setUseDeltaCache(true);
+        store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        assertTrue(store.isDeltaSetCachingEnabled());
+        testManager.cleanBackend(store, StoreType.SYNC);
+
+        Person person = new Person();
+        person.setUsername("DeltaCacheUserName");
+        person.set("CustomField", "CustomValue");
+        Person savedPerson = store.save(person);
+        assertNotNull(savedPerson);
+        assertEquals("DeltaCacheUserName", savedPerson.getUsername());
+
+        store.pushBlocking();
+        Person downloadedPerson = store.pullBlocking(client.query()).getResult().get(0);
+        assertNotNull(downloadedPerson);
+        assertEquals("DeltaCacheUserName", downloadedPerson.getUsername());
+    }
+
+    @Test
+    public void testRead() throws InterruptedException, IOException {
+        client.setUseDeltaCache(true);
+        store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        assertTrue(store.isDeltaSetCachingEnabled());
+        testManager.cleanBackend(store, StoreType.SYNC);
+
+        Person person = new Person();
+        person.setUsername("DeltaCacheUserName");
+        person.set("CustomField", "CustomValue");
+        Person savedPerson = store.save(person);
+        assertNotNull(savedPerson);
+        assertEquals("DeltaCacheUserName", savedPerson.getUsername());
+
+        store.pushBlocking();
+        Person downloadedPerson = store.pullBlocking(client.query()).getResult().get(0);
+        assertNotNull(downloadedPerson);
+        assertEquals("DeltaCacheUserName", downloadedPerson.getUsername());
+
+        List<Person> personList = testManager.find(store, client.query()).getResult();
+        Person person1 = personList.get(0);
+        assertNotNull(person1);
+        assertNotNull(person1.getUsername());
+        assertEquals("DeltaCacheUserName", person1.getUsername());
+    }
+
+    @Test
+    public void testReadALotOfItems() throws InterruptedException, IOException {
+        client.setUseDeltaCache(true);
+        store = DataStore.collection(Person.COLLECTION, Person.class, StoreType.SYNC, client);
+        assertTrue(store.isDeltaSetCachingEnabled());
+        testManager.cleanBackend(store, StoreType.SYNC);
+
+        for (int i = 0; i < 10; i++) {
```

Comment:
I have mentioned this to @vinaygahlawat before as well (for .NET). It will be cleaner to have some common utility functions that prepare data for all tests. Something you should think about as a future improvement.
Something like `createNEntities(int n)`

Reply:
Created in TestManager class